android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt

```diff
@@ -87,15 +94,53 @@ class HomePageViewModel @Inject constructor(
 				it
 			}
 		}
+		/*
+		 * Retrieve LatLng from geocodeApi if addressSearch is defined
+		 */
+		.map { getListingParams ->
+			if (getListingParams.filters.addressSearch != null) {
+				val splitAddress = getListingParams.filters.addressSearch.split(",").toTypedArray()
+				runCatching {
+					runBlocking {
+						geocodeApi.forwardGeocode(
+							addressLine = if (splitAddress.isNotEmpty()) splitAddress[0] else "",
+							addressCity = if (splitAddress.size >= 2) splitAddress[1] else "",
+							addressPostalcode = if (splitAddress.size >= 3) splitAddress[2] else "",
+							addressCountry = if (splitAddress.size >= 4) splitAddress[3] else "Canada",
```

Comment:
I've seen this somewhere before... 🤔 (not important)

Reply:
That's crazy. It's like I copied it from somewhere lol